src/jit/lower.h

```diff
@@ -48,34 +48,67 @@ class Lowering : public Phase
     static void getCastDescription(GenTreePtr treeNode, CastInfo* castInfo);
 
     // LowerRange handles new code that is introduced by or after Lowering.
-    void LowerRange(LIR::Range&& range)
+    void LowerRange(LIR::ReadOnlyRange&& range)
     {
         for (GenTree* newNode : range)
         {
             LowerNode(newNode);
         }
     }
+    void LowerRange(GenTree* firstNode, GenTree* lastNode)
+    {
+        LowerRange(LIR::ReadOnlyRange(firstNode, lastNode));
+    }
 
 private:
+    // ContainCheckRange handles new code that is introduced by or after Lowering,
+    // and that is known to be already in Lowered form.
+    void ContainCheckRange(LIR::Range&& range)
+    {
+        for (GenTree* newNode : range)
+        {
+            ContainCheckNode(newNode);
+        }
+    }
+    void ContainCheckRange(GenTree* firstNode, GenTree* lastNode)
+    {
+        for (GenTree* node = firstNode; node != lastNode->gtNext; node = node->gtNext)
```

Comment:
This could be simply `for (GenTree* node : LIR::ReadOnlyRange(firstNode, lastNode)))`